tests/robustness/engine/engine.go

```diff
@@ -220,9 +221,10 @@ func (e *Engine) formatLogName() string {
 	return fmt.Sprintf("Log_%s", st.Format("2006_01_02_15_04_05"))
 }
 
-// CleanComponents cleans up each component part of the test engine.
-func (e *Engine) CleanComponents() {
-	for _, f := range e.cleanupRoutines {
+// CleanEngineIndexComponents cleans up each component part of the each test engine index.
+func (e *Engine) CleanEngineIndexComponents(idx int) {
```

Comment:
This method will be called once per index.  In each invocation it will invoke
```
os.RemoveAll(e.baseDirPath)
```
Was this intended?

Reply:
I think so, and we can clean up even if one engine fails. As we are returning which will then clean up everything at the end. WDYT?